File: LeetCodeProblemsCMake/Problems/LeetCode3577CountTheNumberOfComputerUnlockingPermutations.cpp
//
// Created by saeed on 12/10/25.
//

#include "LeetCode3577CountTheNumberOfComputerUnlockingPermutations.h"
using namespace std;

int LeetCode3577CountTheNumberOfComputerUnlockingPermutations::countPermutations(vector<int> &complexity) {
    int firstElement = complexity[0];
    for (int i = 1; i < complexity.size(); i++) {
        if (complexity[i] <= firstElement) {
            return 0;
        }
    }
    int mod = 1e9 + 7;
    long long result = 1;
    for (int i = 1; i < complexity.size(); i++) {
        result *= i;
        result %= mod;
    }
    return result;
}

File: LeetCodeProblemsGo/LeetCode3577CountTheNumberOfComputerUnlockingPermutations/LeetCode3577CountTheNumberOfComputerUnlockingPermutations.go
package LeetCode3577CountTheNumberOfComputerUnlockingPermutations

func countPermutations(complexity []int) int {
	firstItem := complexity[0]

	length := len(complexity)
	for i := 1; i < length; i++ {
		if complexity[i] <= firstItem {
			return 0
		}
	}
	var mod int64 = 1e9 + 7
	var result int64 = 1
	for i := 1; i < length; i++ {
		result *= int64(i)
		result %= mod
	}
	return int(result)
}
